x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -247,6 +251,13 @@ public void invalidateAll() {
     public void createApiKey(Authentication authentication, CreateApiKeyRequest request, Set<RoleDescriptor> userRoles,
                              ActionListener<CreateApiKeyResponse> listener) {
         ensureEnabled();
+        if (request.getMetadata() != null && false == request.getMetadata().isEmpty()) {
+            if (securityIndex.getInstallableMappingVersion().before(METADATA_INTRODUCED)) {
```

Comment:
Can we put a bit of logging here ?
```suggestion
            final Version securityMappingVersion = securityIndex.getInstallableMappingVersion() 
            if (securityMappingVersion.before(METADATA_INTRODUCED)) {
                logger.info("The security index [{}] mapping is for version [{}] but API Key metadata requires [{}];"
                    + " the mapping will automatically upgrade to a supported version when the cluster no longer"
                    + " has nodes that are [{}] or earlier", 
                    securityIndex.aliasName(), 
                    securityMappingVersion, 
                    METADATA_INTRODUCED, 
                    FLATTENED_FIELD_TYPE_INTRODUCED);
```

Reply:
Good point. I'd change logging level to `warn` since this branch leads to a failure.